internal/conv/conv.go

```diff
@@ -138,3 +140,36 @@ func BigInt2Uint64Le(z []uint64, x *big.Int) {
 		z[i] = 0
 	}
 }
+
+// MarshalBinary encodes a value into a byte array in a format readable by UnmarshalBinary.
+func MarshalBinary(v cryptobyte.MarshalingValue) ([]byte, error) {
+	var b cryptobyte.Builder
+	b.AddValue(v)
+	return b.Bytes()
+}
+
+// A UnmarshalingValue decodes itself from a cryptobyte.String and advances the pointer.
+// Returns true indicating the reading was successful.
+type UnmarshalingValue interface {
+	ReadValue(*cryptobyte.String) bool
+}
+
+// UnmarshalBinary recovers a value from a byte array.
+// Returns an error if the recovered value is invalid.
+// Any panic raised when calling to ReadValue is recovered and an error is returned instead.
```

Comment:
Why do we need this?

Reply:
this helper function generically implements UnmarshalBinary if there is an implementation of ReadValue (which internally uses the `cryptobyte` package).